doc/examples/collision_detection/lib/main.dart

```diff
@@ -0,0 +1,264 @@
+import 'dart:math';
+
+import 'package:flame/components.dart';
+import 'package:flame/flame.dart';
+import 'package:flame/game.dart';
+import 'package:flame/extensions.dart';
+import 'package:flame/geometry.dart';
+
+import 'package:flutter/material.dart' hide Image, Draggable;
+
+import 'dart:ui';
+
+void main() async {
+  WidgetsFlutterBinding.ensureInitialized();
+  await Flame.device.fullScreen();
+  final game = MyGame();
+  runApp(
+    GameWidget(
+      game: game,
+    ),
+  );
+}
+
+enum Shapes { circle, rectangle, polygon }
+
+abstract class MyCollidable extends PositionComponent
+    with Draggable, Hitbox, Collidable {
+  double rotationSpeed = 0.0;
+  final Vector2 velocity;
+  final delta = Vector2.zero();
+  double angleDelta = 0;
+  bool _isDragged = false;
+  final _activePaint = Paint()..color = Colors.amber;
+  double _wallHitTime = double.infinity;
+
+  MyCollidable(Vector2 position, Vector2 size, this.velocity) {
+    this.position = position;
+    this.size = size;
+    anchor = Anchor.center;
+  }
+
+  @override
+  void update(double dt) {
+    super.update(dt);
+    if (_isDragged) {
+      return;
+    }
+    _wallHitTime += dt;
+    delta.setFrom(velocity * dt);
+    position.add(delta);
+    angleDelta = dt * rotationSpeed;
+    angle = (angle + angleDelta) % (2 * pi);
+  }
+
+  @override
+  void render(Canvas canvas) {
+    super.render(canvas);
+    renderShapes(canvas);
+    final localCenter = (size / 2).toOffset();
+    if (_isDragged) {
+      canvas.drawCircle(localCenter, 5, _activePaint);
+    }
+    if (_wallHitTime < 1.0) {
+      // Show a rectangle in the center for a second if we hit the wall
+      canvas.drawRect(
+        Rect.fromCenter(center: localCenter, width: 10, height: 10),
+        debugPaint,
+      );
+    }
+  }
+
+  @override
+  void onCollision(Set<Vector2> points, Collidable other) {
+    final averageIntersection =
+        points.fold<Vector2>(Vector2.zero(), (sum, v) => sum + v) /
```

Comment:
can `points` be empty?

Reply:
Nope, then it wouldn't be a collision.

Reply:
you can use reduce then :p (it uses the first element as the initial value)

Reply:
Ah, true! :)